### PR TITLE
feat: add group project guide notebook for lesson 08

### PR DESCRIPTION
**Pull request by lvwerra**
Here you go. Let me know if I forgot something or should add more description.

### CODE REVIEW

File: notebooks/lesson08_group-project.ipynb [hunk 1]
```diff
@@ -0,0 +1,949 @@
+{
```
Comment (lewtun) [inline notebooks/lesson08_group-project.ipynb:1]: <ul><li>The convention I've adopted in the other lectures is to only capitalise the first word in section headers.</li><li>Could you include a "References" section and say this is checklist is taken from Appendix B of the Hands-On ML book by A. Geron?</li><li>May 28 is a Thursday, but I am happy for them to submit on Friday the 29th. We should specify the time, e.g. before 23:59:59 CEST so they don't try to sneak in an extra weekend.</li><li>June 4 is a Thursday</li><li>I would drop the classification reference examples to avoid confusing them (we did it last year because there was a mix)</li></ul>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lewtun/dslectures/pull/33/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (lewtun) [inline notebooks/lesson08_group-project.ipynb:1]: <p>Is there something wrong with the last item? It is indented and a "10" when it should be a "9"</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lewtun/dslectures/pull/33/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='4'/>

Comment (lewtun) [inline notebooks/lesson08_group-project.ipynb:1]: <p>I wonder whether one model per team member is a bit tough for them? we could say something like at least 3 models per team, one of which may be a random forest. we might also want to point them to some sklearn links, e.g. linear regression, extratrees, and <a href="https://scikit-learn.org/stable/modules/generated/sklearn.ensemble.HistGradientBoostingRegressor.html" target="_blank">https://scikit-learn.org/stable/modules/generated/sklearn.ensemble.HistGradientBoostingRegressor.html</a></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lewtun/dslectures/pull/33/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

Comment (lewtun) [inline notebooks/lesson08_group-project.ipynb:1]: <p>I have not been using the <code>df_</code>  prefix in previous lectures as i decided it is bad practice :) i would use just <code>train</code>  and <code>test</code>  or <code>train_data</code>  etc</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lewtun/dslectures/pull/33/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='15'/>

Comment (lewtun) [inline notebooks/lesson08_group-project.ipynb:1]: <p>instead of saying "Fit" and "Predict" would it be simpler to just use the sample submission file as follows:</p><p> </p><pre class="ql-syntax" spellcheck="false">preds = train['price'].median()

submission = pd.read_csv('sample_submission.csv')
submission['Predicted'] = preds

etc etc
</pre><p> </p><p>this would allow us to avoid the copy and rename step</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lewtun/dslectures/pull/33/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='18'/>

Comment (lvwerra) [reply notebooks/lesson08_group-project.ipynb:1]: <p>can you explain? i only see a 9..</p>

Comment (lvwerra) [reply notebooks/lesson08_group-project.ipynb:1]: <p>good catch with the dates</p>

Comment (lvwerra) [reply notebooks/lesson08_group-project.ipynb:1]: <p>My thinking was that this way there would be one part of the notebook that would be unique to the team members. </p>

Comment (lvwerra) [reply notebooks/lesson08_group-project.ipynb:1]: <p>did you also stop import numpy as np?</p><p> </p>

Comment (lvwerra) [reply notebooks/lesson08_group-project.ipynb:1]: <p>I changed it a bit and used your approach to fill the submission file. However, I left the fit/predict step in because I thought it would make it clearer what they should do with the train and test set. If you think it is overkill I can delete it.</p>

Comment (lewtun) [reply notebooks/lesson08_group-project.ipynb:1]: <p>it must be a bug in reviewNB then. don't worry about it</p><p> </p>

Comment (lewtun) [reply notebooks/lesson08_group-project.ipynb:1]: <p>i see, so how about we </p><p> </p><p>a) point them to suitable models to play with (i.e. enough choices for each group, probably 4 models max)</p><p>b) rephrase the test to "train at least one model per team member. You can only have one Random Forest model per team, so you may want to investigate the following alternatives for regression: ..."</p>

Comment (lewtun) [reply notebooks/lesson08_group-project.ipynb:1]: <p>are you teasing me :) </p><p> </p><p>using aliases is fine, but adding data types to objects is unnecessary IMO</p>

File: notebooks/lesson08_group-project.ipynb [hunk 2]
```diff
@@ -0,0 +1,1055 @@
...
```
Comment (lewtun) [inline notebooks/lesson08_group-project.ipynb:1]: <p>it occurred to me that we should add a subsection here with "Deliverables" and list the single jupyter notebook + video presentation + slides. I know we says this elsewhere, but it might help to collect it in one place</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lewtun/dslectures/pull/33/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (lewtun) [inline notebooks/lesson08_group-project.ipynb:1]: <p>should we mention here that their presentation should be a 15 minute video with slides?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lewtun/dslectures/pull/33/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='8'/>

Comment (lewtun) [inline notebooks/lesson08_group-project.ipynb:1]: <p>i still think this step is not needed because they will eventually do something like </p><p> </p><pre class="ql-syntax" spellcheck="false">preds = model.fit(test)
</pre><p> </p><p>and then can inject the preds into the submission dataframe. adding the preds as a column to the test dataframe feels a bit unusual to me</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lewtun/dslectures/pull/33/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='23'/>
